src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx

```diff
@@ -5,18 +5,18 @@ import {DisplayAction} from '@jahia/ui-extender';
 import {useSelector} from 'react-redux';
 import styles from './MainActionBar.scss';
 
-const ButtonRenderer = getButtonRenderer();
 const ButtonRendererShortLabel = getButtonRenderer({labelStyle: 'short'});
 const ButtonRendererNoLabel = getButtonRenderer({labelStyle: 'none'});
 
 export const MainActionBar = () => {
-    const {path} = useSelector(state => ({path: state.jcontent.path}));
-
+    const {path, selection} = useSelector(state => ({path: state.jcontent.path, selection: state.jcontent.selection}));
+    let editPath = (selection && selection.length === 0) ? path : selection[0];
+    let enabled = selection.length < 2;
     return (
         <div className={styles.root}>
             <Separator variant="vertical"/>
-            <DisplayAction actionKey="pageComposer" context={{path}} render={ButtonRenderer} size="big" variant="ghost" color="accent" className={styles.item}/>
-            <DisplayAction actionKey="edit" context={{path}} render={ButtonRenderer} size="big" variant="outlined" className={styles.item}/>
+            <DisplayAction actionKey="pageComposer" context={{path: editPath}} render={getButtonRenderer({enabled: enabled})} size="big" variant="ghost" color="accent" className={styles.item}/>
+            <DisplayAction actionKey="edit" context={{path: editPath, enabled: enabled}} render={getButtonRenderer({enabled: enabled})} size="big" variant="outlined" className={styles.item}/>
```

Comment:
Enabled status should be handled by the action

Reply:
I know but I don't have access to the selection from content-editor side or page-composer side